### PR TITLE
Add interface to verify trusted MailServers

### PR DESCRIPTION
**Pull request by adambabik**
The goal of this PR is to add an interface to verify MailServers. In this PR, MailServers are hardcoded in status-go. The next iteration will use a smart contract.

### CODE REVIEW

**Review by gravityblast — APPROVED** [review]
LGTM

**Comment by gravityblast** [comment]
@adambabik I see CI is failing with `The contents of vendor differ after "dep ensure":`

**Review by dshulyak — APPROVED** [review]

File: peers/cotopicpool.go [hunk 1]
```diff
@@ -45,14 +54,34 @@ var sendEnodeDiscovered = signal.SendEnodeDiscovered
 // ConfirmAdded calls base TopicPool ConfirmAdded method and sends a signal
 // confirming the enode has been discovered.
 func (t *cacheOnlyTopicPool) ConfirmAdded(server *p2p.Server, nodeID discover.NodeID) {
-	t.TopicPool.ConfirmAdded(server, nodeID)
-	sendEnodeDiscovered(nodeID.String(), string(t.topic))
+	trusted := t.verifier.VerifyNode(context.TODO(), nodeID)
```
Comment (divan) [inline peers/cotopicpool.go:57]: Okay, this is probably related to the CacheOnlyTopicPool design, and not this PR in particular, but from the code it's easy to get the assumption that `cacheOnlyTopicPool` is just a topic pool without persistence. But in the `PeerPool` code it becomes clear that `cacheOnlyTopicPool` is a special pool for mailservers.

I think this information (that it's a special pool for mailservers) should be somehow represented in types/naming. If we'll want someday to use this cache-only topic pool for other purposes, this verifier code will not make sense here anymore.

Does it make sense, or am I missing something about this cotopicpool type?

File: peers/verifier/verifier.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	for _, peer := range peers {
+		knownPeers[peer] = true
+	}
+
+	return &LocalVerifier{KnownPeers: knownPeers}
+}
+
+// VerifyNode checks if a given node is trusted using a local list.
+func (v *LocalVerifier) VerifyNode(_ context.Context, nodeID discover.NodeID) bool {
+	if res, ok := v.KnownPeers[nodeID]; ok {
```
Comment (divan) [inline peers/verifier/verifier.go]: could be just

```go
_, ok := v.KnownPeers[nodeID];
return ok
```
(`bool` could be replaced by `struct{}` then)

File: peers/cotopicpool.go (hunk 1 above)
Comment (adambabik) [reply peers/cotopicpool.go:57]: You are right. Overall, the whole design was built with a single type of `TopicPool` in mind but later it turned out that there might be more of them needed. 

A refactor idea is to have a very basic `TopicPool` that doesn't do much and more specialized topic pools built on top of it. Currently, `TopicPool` has all features and if you want a topic pool with fewer features, you have a problem as you would need to write it from scratch duplicating the code.

**Pull request merged**